File: src/main/java/com/codeperfector/examples/kafkaconsumer/AppConfig.java
package com.codeperfector.examples.kafkaconsumer;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.util.HashMap;
import java.util.List;
import java.util.Map;


@Component
@Getter
@Setter
@ConfigurationProperties(prefix = "kafka")
public class AppConfig {

    public List<String> topics;
    public Map<String, String> consumer = new HashMap<>();
    public Map<String, String> producer = new HashMap<>();
}

File: src/main/java/com/codeperfector/examples/kafkaconsumer/AppConfig.kt
package com.codeperfector.examples.kafkaconsumer

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component


@Component
@ConfigurationProperties(prefix = "kafka")
class AppConfig {
    var topics: MutableList<String>? = null
    var consumer: MutableMap<String?, String?> = mutableMapOf()
    var producer: MutableMap<String?, String?> = mutableMapOf()
}